pkg/operator/defaultscccontroller/defaultscccontroller.go

```diff
@@ -0,0 +1,141 @@
+package defaultscccontroller
+
+import (
+	"fmt"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	securityv1 "github.com/openshift/api/security/v1"
+	securityv1client "github.com/openshift/client-go/security/clientset/versioned/typed/security/v1"
+	securityv1listers "github.com/openshift/client-go/security/listers/security/v1"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+)
+
+const (
+	ControllerName = "default-scc-upgradeable"
+)
+
+type Options struct {
+	Config         *rest.Config
+	OperatorClient v1helpers.OperatorClient
+	Recorder       events.Recorder
+	ResyncPeriod   time.Duration
+}
+
+func NewDefaultSCCController(options *Options) (controller *DefaultSCCController, err error) {
+	client, clientErr := securityv1client.NewForConfig(options.Config)
+	if clientErr != nil {
+		err = fmt.Errorf("[%s] failed to create client for security/v1 - %s", ControllerName, clientErr.Error())
+		return
+	}
+
+	defaultSCCSet, renderErr := NewDefaultSCCCache()
```

Comment:
why these special error var names? Any shadowing logic going on here?

Reply:
No shadowing, the same explanation I gave above.